sys/auto_init/wdt_thread/wdt.c

```diff
@@ -36,12 +37,20 @@
 
 static char WORD_ALIGNED wdt_stack[WDT_THREAD_STACKSIZE];
 
+static bool _wdt_enabled = true;
+
+void wdt_thread_stop(void)
+{
+    _wdt_enabled = false;
+    wdt_stop();
+}
+
 static void *_wdt_thread(void *ctx)
 {
     (void)ctx;
     unsigned sleep_ms = (CONFIG_PERIPH_WDT_WIN_MIN_MS + CONFIG_PERIPH_WDT_WIN_MAX_MS)
                       / 2;
-    while (1) {
+    while (_wdt_enabled) {
         ztimer_sleep(ZTIMER_MSEC, sleep_ms);
         wdt_kick();
```

Comment:
```suggestion
    while (1) {
        ztimer_sleep(ZTIMER_MSEC, sleep_ms);
        if (!_wdt_enabled) {
            break;
        }
        wdt_kick();
```
to avoid kicking the watchdog when it's already disabled